packages/recommend-react/src/useRecommendations.ts

```diff
@@ -19,35 +20,48 @@ export function useRecommendations<TObject>({
   recommendClient,
   threshold,
   transformItems: userTransformItems,
-}: GetRecommendationsProps<TObject>) {
-  const [result, setResult] = useState<GetRecommendationsResult<TObject>>({
-    recommendations: [],
-  });
+  initialState: userInitialState,
+}: GetRecommendationsProps<TObject> & {
+  initialState?: InitialResults<TObject>;
+}) {
+  const isFirstRenderRef = useRef(false);
```

Comment:
Technically we're in the first render the first time this executes.

```suggestion
  const isFirstRenderRef = useRef(true);
```

Reply:
✅